src/Hl7.Fhir.ElementModel/Types/Date.cs

```diff
@@ -137,6 +137,33 @@ private static bool tryParse(string representation, out Date value)
             return success;
         }
 
+        public static Date operator +(Date me, Quantity value)
+        {
+            var dto = me._parsedValue;
+            switch (value.Unit)
+            {
+                // we can ignore precision, as the precision will "trim" it anyway, and if we add 13 months, then the year can tick over nicely
+                case "a": // UCUM
+                case "years":
+                case "year":
+                    dto = me._parsedValue.AddYears((int)value.Value);
+                    break;
+                case "mo": // UCUM
+                case "month":
+                case "months":
+                    dto = me._parsedValue.AddMonths((int)value.Value);
+                    break;
+                case "d": // UCUM
```

Comment:
I think you've misread that part, but will add week.
The notes in the spec state that there are calendar units.

Reply:
6.7 does not mention these UCUM units at all.  It actually says:

> To avoid the potential confusion of calendar-based date/time arithmetic with definite duration date/time arithmetic, FHIRPath defines definite-duration date/time arithmetic for seconds and below, and calendar-based date/time arithmetic for seconds and above. At the second, calendar-based and definite-duration-based date/time arithmetic are identical. **If a definite-quantity duration above seconds appears in a date/time arithmetic calculation, the evaluation will end and signal an error to the calling environment.**

Reply:
The spec has explicitly described that '1 year' is not equal to '1 a', although it is equivalent `'1 year' ~ '1 a'`, but for the date/time calculations, it seems to explicitly forbid the use.

Reply:
I'll find why I put it in and get pack to you.

Reply:
As noted above, that's in the fhir spec fhirpath page, not fhirpath itself.

Reply:
@ewoutkramer I've done the day/month converstions as they are documented in the spec, but there's no "conversion" for the weeks, should I treat it as though it was days?

Reply:
Yes, let's do that.

Reply:
How would this work for you?
```
       case "weeks":
           if (dateValue.Precision == DateTimePrecision.Year)
                dto = dateValue._parsedValue.AddYears((int)(addValue.Value / 52));
           else if (dateValue.Precision == DateTimePrecision.Month)
                dto = dateValue._parsedValue.AddMonths((int)(addValue.Value * 7 / 30));
            else
                dto = dateValue._parsedValue.AddDays(((int)addValue.Value) * 7);
            break;
```
52 weeks in a year, and when at month level, assume was talking about average days in a month, or should it be more like `AddMonths((int)(addValue.Value/52*12))` so double based multiplication and division, then the cast trims the rounding to the values requested precision.

Reply:
I think that reflects what Grahame has written. I guess this makes the unit test that he provided work?

Reply:
Cool, I'll recheck that there are enough unit test cases covering all these combinations.
(existing ones pass)
And wait on an answer tomorrow on which way the fhir chat points it.